### ISSUE TITLE
[Bug] BlockResourcesPlugin does not work alongside with AdblockerPlugin.

### ISSUE THREAD

**Issue by Aex12** [berstend/puppeteer-extra#585]
**Describe the bug**
Whenever you try to enable both plugins alongside, BlockResourcesPlugin stops working. I've set up to block images, stylesheets and media types, and it works fine when AdblockerPlugin is not enabled, but if you enable both, BlockResourcesPlugin stop blocking the provided resources.

**Code Snippet**

```javascript
const puppeteer = require('puppeteer-extra');
const StealthPlugin = require('puppeteer-extra-plugin-stealth');
const BlockResourcesPlugin = require('puppeteer-extra-plugin-block-resources');
const AdblockerPlugin = require('puppeteer-extra-plugin-adblocker');

puppeteer.use(StealthPlugin());
puppeteer.use(AdblockerPlugin());
puppeteer.use(BlockResourcesPlugin({ 
	blockedTypes: new Set(['image', 'stylesheet', 'media']),
}));

(async () => {
  const browser = await puppeteer.launch({ headless: true });
  const page = await browser.newPage();
  await page.goto('https://www.google.com/');
  await page.screenshot({ path: 'example.png' });

  await browser.close();
})()
```
**Steps to reproduce**
After running the above code snippet, check the example.png image created to see that styles and images are in fact loaded. If we remove AdblockerPlugin from the equation, the screenshot is as expected, without styles nor images.

**Versions**

  System:
    OS: Linux 5.14 Fedora 34 (Thirty Four) 34 (Thirty Four)
    CPU: (16) x64 AMD Ryzen 7 3800X 8-Core Processor
    Memory: 1.95 GB / 15.54 GB
    Container: Yes
    Shell: 5.1.0 - /bin/bash
  Binaries:
    Node: 14.17.4 - ~/.nvm/versions/node/v14.17.4/bin/node
    Yarn: 1.22.15 - ~/.nvm/versions/node/v14.17.4/bin/yarn
    npm: 6.14.14 - ~/.nvm/versions/node/v14.17.4/bin/npm
  npmPackages:
    puppeteer: ^10.4.0 => 10.4.0 
    puppeteer-extra: ^3.2.3 => 3.2.3 
    puppeteer-extra-plugin-adblocker: ^2.12.0 => 2.12.0 
    puppeteer-extra-plugin-block-resources: ^2.3.0 => 2.3.0 
    puppeteer-extra-plugin-stealth: ^2.9.0 => 2.9.0

**Comment by firmart** [comment]
The solution is addressed there: https://github.com/berstend/puppeteer-extra/issues/156#issuecomment-757049953.

**Comment by MeikyuuTrader** [comment]
I'm getting a request already handled error when enabling puppeteer page proxy and both adblocker plugin and block resources. But everything works fine if I disabled the adblocker.  

```
/node_modules/puppeteer/lib/cjs/puppeteer/common/assert.js:26
        throw new Error(message);
              ^

Error: Request is already handled!
    at assert (/node_modules/puppeteer/lib/cjs/puppeteer/common/assert.js:26:15)
    at HTTPRequest.abort (/node_modules/puppeteer/lib/cjs/puppeteer/common/HTTPRequest.js:453:32)
    at requestHandler (/node_modules/puppeteer-page-proxy/src/core/proxy.js:41:23)
    at processTicksAndRejections (node:internal/process/task_queues:96:5)
    at async $ppp_request_listener (/node_modules/puppeteer-page-proxy/src/core/proxy.js:82:9)
    at async HTTPRequest.finalizeInterceptions (/node_modules/puppeteer/lib/cjs/puppeteer/common/HTTPRequest.js:151:9)
```

    "puppeteer": "^13.7.0",
    "puppeteer-extra": "^3.2.3",
    "puppeteer-extra-plugin-adblocker": "^2.12.0",
    "puppeteer-extra-plugin-block-resources": "^2.3.0",
    "puppeteer-extra-plugin-stealth": "^2.9.0",
    "puppeteer-page-proxy": "^1.2.8"

**Comment by MeikyuuTrader** [comment]
See this solution by SnowLew [#5334](https://github.com/puppeteer/puppeteer/issues/5334#issuecomment-1046106755) to use BlockResources and have Adblocker working. 

Effectively you can't use the BlockResourcesPlugin, you need to manually remove the existing request listener and call another request listener to block resources.

**Comment by berstend** [comment]
This is fixed in #592 :)
